editor/src/core/model/element-template-utils.spec.tsx

```diff
@@ -739,3 +749,301 @@ describe('rearrangeJsxChildren', () => {
+    ])
+  })
+
+  it('project with arbitrary js block that has a conditional expression inside', () => {
+    const projectFile = getParseSuccessForStoryboardCode(
+      makeTestProjectCodeWithSnippet(`
+        <div style={{ ...props.style }} data-uid='aaa'>
+          <div data-uid='parent' >
+            <div data-uid='child-d' />
+            {[1, 2, 3].map(id => {
+              return (
+                <div data-uid='mapped-root'>
+                  <div data-uid='mapped-child' />
+                </div> 
+              )
+            })}
+            {[1, 2, 3].map(id => {
+              return (
+                true ? 
+                  (
+                    <div data-uid='ternary-true-root'>
+                      <div data-uid='ternary-true-child' />
+                    </div> 
+                  ) : (
+                    <div data-uid='ternary-false-root'>
+                      <div data-uid='ternary-false-child' />
+                    </div> 
+                  )
+              )
+            })}
+            <div data-uid='child-b' />
+            <div data-uid='child-a' />
+          </div>
+        </div>
+      `),
+    )
+
+    expectElementAtPathHasMatchingUIDForPaths(projectFile, [
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/mapped-root',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/mapped-root/mapped-child',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ternary-true-root',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ternary-true-root/ternary-true-child',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ternary-false-root',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ternary-false-root/ternary-false-child',
+    ])
+  })
+
+  it('conditional expressions', () => {
+    const projectFile = getParseSuccessForStoryboardCode(
+      makeTestProjectCodeWithSnippet(`
+        <div style={{ ...props.style }} data-uid='aaa'>
+          <div data-uid='parent' >
+            <div data-uid='child-d' />
+            {true ? 
+              (
+                <div data-uid='ternary-true-root'>
+                  <div data-uid='ternary-true-child' />
+                </div> 
+              ) : (
+                <div data-uid='ternary-false-root'>
+                  <div data-uid='ternary-false-child' />
+                </div> 
+              )
+            }
+            <div data-uid='child-b' />
+            <div data-uid='child-a' />
+          </div>
+        </div>
+      `),
+    )
+
+    expectElementAtPathHasMatchingUIDForPaths(projectFile, [
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ca0',
```

Comment:
I'd recommend using `FOR_TESTS_setNextGeneratedUids` or similar to make the conditional expression have a fixed UID.

Reply:
fixed!